velociraptor/observations/__init__.py

```diff
@@ -29,7 +38,72 @@ def load_observation(filename: str):
         Observational data instance read from file.
     """
 
+    warn(
+        "load_observation is deprecated and will be removed in a future version. "
+        "Please use load_observations that can load data from multi-redshift "
+        "datasets introduced in velociraptor v0.12.0.",
+        DeprecationWarning,
+    )
+
     data = ObservationalData()
     data.load(filename)
 
     return data
+
+
+def load_observations(
+    filenames: Union[str, Iterable[str]], redshift_bracket: List[float] = [0.0, 1000.0]
+):
+    """
+    Parameters
+    ----------
+
+    filename: str, Iterable[str]
+        Filename(s) of the observational dataset that you wish to load.
+        Should probably end in .hdf5. See the documentation for
+        :class:`velociraptor.observations.objects.ObservationalData`
+        and :class:`velociraptor.observations.objects.MultiRedshiftObservationalData`
+        for more information.
+
+    redshift_bracket: str
+        Redshift bracket to overlap with. If any of the observations in the
+        file overlap with this bracket, they are returned. By default, this
+        bracket is 0.0 to 1000.0, so will encompass all reasonable
+        observations present in the file.
+    
+
+    Returns
+    -------
+
+    List[velociraptor.observations.objects.ObservationalData]:
+        Observational data instances read from file that overlap with your
+        specified redshift bracket.
+    """
+
+    returned_data = []
+
+    if not isinstance(filenames, list):
+        filenames = [filenames]
+
+    for filename in filenames:
+        try:
+            multi_z = MultiRedshiftObservationalData()
+            multi_z.load(filename)
+
+            returned_data += multi_z.get_datasets_overlapping_with(
+                redshifts=redshift_bracket
+            )
+        except ObservationalDataError:
+            data = ObservationalData()
+            data.load(filename)
+
+            lower, upper = redshift_bracket
```

Comment:
If an observational-data file has no lower and upper redshift bounds, VR library will crash while executing `multi_z.load(filename)` with an error

> KeyError: "Can't open attribute (can't locate attribute: 'redshift_lower')"

@JBorrow Could you also add a comment here indicating that we have to add the lower and upper z bounds because the file does not have them?

Reply:
That's not what this does; the redshift_bracket is used to determine whether or not this file should be included and plotted, unless I'm misunderstanding what you've said?

Reply:
Ah, I see. I confused `redshift_bracket`, which is a **property of the plot**, with that of the observational data ( `data.redshift_lower`  and `data.redshift_upper` )

Reply:
How about we change `lower, upper` to `figure_redshift_lower, figure_redshift_upper` to make it easier to distinguish from `data.redshift_lower` and `data.redshift_upper`?